sgt-back-end/index.js

```diff
@@ -0,0 +1,184 @@
+});
+
+const app = express();
+const jsonMiddleWare = express.json();
+
+app.use(jsonMiddleWare);
+
+app.get('/api/grades', (req, res) => {
+
+  const sql = 'select * from grades';
+
+  db.query(sql)
+    .then(result => {
+      const grades = result.rows;
+      res.json(grades);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.post('/api/grades', (req, res) => {
+  const sql = `insert into "grades" ("name", "course", "score")
+                  values ($1, $2, $3)
+                  returning *`;
+
+  const { name, course, grade } = req.body;
+
+  if (name === undefined || course === undefined || grade === undefined) {
+    res.status(400).send({
+      error: 'The name, course and grade fields are required'
+    });
+    return;
+  }
+
+  // const score = parseInt(grade);
+  if (!Number.isInteger(grade) || grade < 0 || grade > 100) {
+    res.status(400).send({
+      error: 'The grade must be a number from 0 to 100'
+    });
+    return;
+  }
+
+  const params = [name, course, grade];
+  db.query(sql, params)
+    .then(result => {
+      res.status(201).send(result.rows[0]);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.put('/api/grades/:gradeId', (req, res) => {
+
+  const sql = `update "grades"
+                set "name" = $1, "course" = $2, "score" = $3
+                where "gradeId" = $4
+                returning *`;
+
+  const idSql = 'select "gradeId" from "grades"';
+
+  const { name, course, grade } = req.body;
+  const gradeId = parseInt(req.params.gradeId);
+
+  if (name === undefined || course === undefined || grade === undefined) {
+    res.status(400).send({
+      error: 'The name, course and grade fields are required'
+    });
+    return;
+  }
+
+  const score = parseInt(grade);
+  if (!Number.isInteger(score) || score < 0 || score > 100) {
+    res.status(400).send({
+      error: 'The grade must be a number from 0 to 100'
+    });
+    return;
+  }
+
+  let idIsValid = false;
+
+  db.query(idSql)
+    .then(result => {
+      for (let i = 0; i < result.rows.length; i++) {
+        if (result.rows[i].gradeId === gradeId) {
+          idIsValid = true;
+          break;
+        }
+      }
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+
+  const params = [name, course, score, gradeId];
+
+  db.query(sql, params)
+    .then(result => {
+      if (idIsValid) {
+        res.json(result.rows[0]);
+      } else {
+        res.status(404).json({
+          error: `Cannot find grade with gradeId ${gradeId}`
+        });
+      }
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
```

Comment:
Instead of querying the database twice, you can include a `returning` clause in your `update` statement. If nothing actually got updated, `result.rows` will be empty for the update. In general, it's never a good idea to grab the universe and loop through it to do a check.